### ISSUE TITLE
Navigation controls can't identify last slide when using perView: 'auto'

### ISSUE THREAD

**Issue by Omitat** [rcbyr/keen-slider#214]
Hello.

I am having an issue identifying when I have reached the end of the available slide panels when using `slides: {perView: 'auto'}`, fixed widths on slide items and navigational controls ( < > ) to move the slides left or right. 

When using the navigational controls the relative index tends to be 1 - 2 indexes from the last slide in the carousel. This causes odd behavior when disabling/enabling the button across various viewports as the tracked index is often not relative to the final slide (the relative index is not the final slide index and progress does not hit 1).

Here's the [Sandbox](https://codesandbox.io/s/stoic-monad-9shx8f).

Any help would be appreciated, thanks!

**Comment by fennadew** [comment]
I have the same problem. I tried to use the slideChanged function 
```
    slideChanged: slider => {
      setFirstSlide(slider.track.details.progress === 0)
      setLastSlide(slider.track.details.progress === 1)
    },
```

But the function seems to fire before the progress is correct. Definitely seems like a bug to me.  I moved the code to `detailsChanged` which is not ideal because it fires often & there is some delay in hiding/showing the nav.

**Comment by rcbyr** [comment]
Hey @Omitat, Hey @fennadew 

I thought about the problem and made an adjustment to the maxIdx property of the track details. It now also gives the maximum reachable idx of a slider that is not in loop mode. I updated the codesandbox and hope it helps: https://codesandbox.io/s/mutable-waterfall-lhhzq3?file=/src/App.js

**Comment by boomikuma** [comment]
> Hey @Omitat, Hey @fennadew
> 
> I thought about the problem and made an adjustment to the maxIdx property of the track details. (v6.6.7) It now also gives the maximum reachable idx of a slider that is not in loop mode. I updated the codesandbox and hope it helps: https://codesandbox.io/s/mutable-waterfall-lhhzq3?file=/src/App.js

It works! However, I think there's still a bug; it seems that when the slide is not full, `maxIdx` will show as the total number of slides instead of 0. For example, go to https://codesandbox.io/s/mutable-waterfall-lhhzq3?file=/src/App.js and change all the `maxWidth` and `minWidth` to small number. 

Look at `maxIdx` in console 
<img width="1251" alt="image" src="https://user-images.githubusercontent.com/37561295/165998042-a0e62d5b-5bf1-4994-9f93-f80343951cab.png">

If we increase the width of each slide so that they overflow, the `maxIdx` will display correctly:
<img width="1257" alt="image" src="https://user-images.githubusercontent.com/37561295/165998422-9925c9a4-1f84-440a-9a30-0637a578293c.png">

**Comment by rcbyr** [comment]
Thank you for pointing this out to me. Fixed in v6.6.11
